File: examples/js/src/005-git-commit-message.ts
import { createAgent } from "langchain";
import { ChatAnthropic } from "@langchain/anthropic";
import executeCommand from "./tools/executeCommand";

const model = new ChatAnthropic({
  model: "claude-sonnet-4-6",
});

const agent = createAgent({
  model: model,
  tools: [executeCommand],
});

async function main() {
  const response = await agent.invoke({
    messages: [
      {
        role: "user",
        content:
          "Please generate a commit message for the current changes in the current directory using the provided tools.",
      },
    ],
  });

  console.log(response.messages[response.messages.length - 1].content);
}

main();

File: examples/python/src/005_git_commit_message.py
import asyncio
from langchain_anthropic import ChatAnthropic
from langchain.agents import create_agent
from tools.execute_command import execute

model = ChatAnthropic(model="claude-opus-4-8")

agent = create_agent(model=model, tools=[execute])


async def main():
    response = await agent.ainvoke({
        "messages": [
            {
                "role": "user",
                "content": "Please generate a commit message for the current changes in the current directory using the provided tools.",
            }
        ]
    })
    print(response["messages"][-1].content)


asyncio.run(main())
